packages/next-app/src/app/internalmanager/(components)/TokenSelect.tsx

```diff
@@ -156,15 +172,100 @@ function TokenModal({
     }
   }
 
+  const tokenExplorerUrl = buildExplorerTokenURL({
+    chainId: chain!.id,
+    tokenAddress: tokenSearch.toLowerCase() as `0x${string}`,
+  });
+
   useEffect(() => {
     if (!internalBalanceData?.user?.userInternalBalances) return;
     getWalletBalance(tokenAdresses as `0x${string}`[]);
   }, [internalBalanceData]);
 
+  useEffect(() => {
+    if (addressRegex.test(tokenSearch)) {
```

Comment:
`#would` (style) return early if the condition is not met to avoid indenting multiple conditions. also that follows the same pattern you used in lines 218-223